### PR TITLE
[StructuralMechanicsApplication] Fixing impose_rigid_movement_process

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixing not reference/pointer used on variables in impose_rigid_movement_process.  Minor clean done too

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp [hunk 1]
```diff
@@ -60,37 +60,37 @@ void ImposeRigidMovementProcess::ExecuteInitialize()
...
-    std::vector<VariableComponent<ComponentType>> master_components_list_variables, slave_components_list_variables;
+    std::vector<Variable<double>*> master_double_list_variables, slave_double_list_variables;
+    std::vector<VariableComponent<ComponentType>*> master_components_list_variables, slave_components_list_variables;
     const std::string& master_variable_name = mThisParameters["master_variable_name"].GetString();
     // The master variable
     if(KratosComponents<Variable<double>>::Has(master_variable_name)){
-        Variable<double> variable = KratosComponents<Variable<double>>::Get(master_variable_name);
-        master_double_list_variables.push_back(variable);
+        const Variable<double>& r_variable = KratosComponents<Variable<double>>::Get(master_variable_name);
+        master_double_list_variables.push_back(&const_cast<Variable<double>&>(r_variable));
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp]: could you get rid of the `const_cast` ...

File: applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp [hunk 2]
```diff
@@ -60,37 +60,37 @@ void ImposeRigidMovementProcess::ExecuteInitialize()
...
-        constrain.SetId(constraint_id);
+    for (auto& r_constrain : r_root_model_part.MasterSlaveConstraints()) {
+        r_constrain.SetId(constraint_id);
         ++constraint_id;
     }
 
     // Getting list of variables
-    std::vector<Variable<double>> master_double_list_variables, slave_double_list_variables;
-    std::vector<VariableComponent<ComponentType>> master_components_list_variables, slave_components_list_variables;
+    std::vector<Variable<double>*> master_double_list_variables, slave_double_list_variables;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp]:
```suggestion
    std::vector<const Variable<double>*> master_double_list_variables, slave_double_list_variables;
```

... with this?

**Review by philbucher — COMMENTED** [review]
comment regarding `const_cast`

File: applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp]: I don't know if it works, I will try

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp]: (and fix other thing in the way)

**Review by philbucher — APPROVED** [review]

**Pull request merged**